src/dex-ui/pages/CreateProposal/ProposalForms/ContractUpgradeProposalForm.tsx

```diff
@@ -1,16 +1,123 @@
-import { FormControl, Input } from "@chakra-ui/react";
+import { FormControl, Input, FormErrorMessage, Flex, Button, Spacer, Text } from "@chakra-ui/react";
+import { ReactElement, useState } from "react";
+import { useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+import { TextEditor } from "../../../../dex-ui-components";
+import { useDexContext } from "../../../hooks";
+import { ProposalType } from "../../../store/governanceSlice";
+import { FileUploader } from "../../../../dex-ui-components";
+
+interface ContractUpgradeProposalFormData {
+  title: string;
+  description: string;
+  linkToDiscussion: string;
+  abiFile: string;
+  abiFileName: string;
+  newImplementationAddress: string;
+  proxyAddress: string;
+}
+
+export function ContractUpgradeProposalForm(): ReactElement {
+  const { governance } = useDexContext(({ governance }) => ({ governance }));
+  const navigate = useNavigate();
+  const {
+    handleSubmit,
+    register,
+    formState: { errors, isSubmitting },
+  } = useForm<ContractUpgradeProposalFormData>();
+
+  const handleCancelClick = () => navigate("/governance");
+
+  const onFileReadAction = (file: string, fileName: string) => {
+    console.log("File", file);
+    setAddress(fileName);
+  };
+  async function onSubmit(data: ContractUpgradeProposalFormData) {
+    // TODO: Call the Contract
+  }
+
+  const [address, setAddress] = useState("");
 
-/**
- * TODO: Update input fields to match full proposal creation feature set.
- * Use the TokenTransferProposalForm component as a reference for how to use Chakra
- * components and react-form-hooks to create a form.
- */
-export function ContractUpgradeProposalForm() {
   return (
-    <form>
-      <FormControl>
-        <Input variant="form-input" id="title" placeholder="Proposal Title" />
-      </FormControl>
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <Flex direction="column" gap="0.5rem">
+        <FormControl isInvalid={Boolean(errors.title)}>
+          <Input
+            variant="form-input"
+            id="title"
+            placeholder="Proposal Title"
+            {...register("title", {
+              required: { value: true, message: "A title is required." },
+            })}
+          />
+          <FormErrorMessage>{errors.title && errors.title.message}</FormErrorMessage>
+        </FormControl>
+        <FormControl>
+          <TextEditor id="description" placeholder="Description" {...register("description")} />
+        </FormControl>
+        <FormControl>
+          <Input
+            variant="form-input"
+            id="linkToDiscussion"
+            placeholder="Link to Discussion (optional)"
+            {...register("linkToDiscussion")}
+          />
+        </FormControl>
+        <FormControl>
+          <FileUploader
```

Comment:
I know this wasn't specified in the wireframes but can we update the hover cursor type to `pointer` for the clickable area? I left a comment in Figma for Anna to confirm that this is the preferred cursor type for this component.

Reply:
This is done in previous PR